src/modules/esl-utils/async/promise.ts

```diff
@@ -1,11 +1,11 @@
 import type {AnyToAnyFnSignature} from '../misc/functions';
 
 /**
- * @returns Promise that will be resolved in `timeout` with optional `payload`
+ * @returns Promise that will be resolved in `timeout` with optional `payload` if `isReject` is true, and will be rejected if `isReject` is false
```

Comment:
```suggestion
 * @returns Promise that will be resolved or rejected in `timeout` with an optional `payload`
 * If `isReject` is `true` the result promise will be rejected, otherwise (by default) the result promise will be resolved
```